rrule/src/core/properties.rs

```diff
@@ -291,3 +338,146 @@ impl RRuleProperties {
         })
     }
 }
+
+impl FromStr for RRuleProperties {
+    type Err = ParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        parse_rule(s)
+    }
+}
+
+impl Display for RRuleProperties {
+    /// Generates a string based on the [iCalendar RRULE spec](https://datatracker.ietf.org/doc/html/rfc5545#section-3.8.5.3).
+    /// It doesn't prepend "RRULE:" to the string.
+    /// This function doesn't validate the existing object and may generate an invalid string like 'FREQ=YEARLY;INTERVAL=-1'
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let mut res = format!("FREQ={};", &self.freq);
+
+        if let Some(until) = &self.until {
+            res.push_str(&format!("UNTIL={};", until.format("%Y%m%dT%H%M%SZ")));
+        }
+
+        if let Some(count) = &self.count {
+            res.push_str(&format!("COUNT={};", count));
+        }
+
+        // One interval is the default, no need to expose it.
+        if self.interval != 1 {
+            res.push_str(&format!("INTERVAL={};", &self.interval));
+        }
+
+        // Monday is the default, no need to expose it.
+        if self.week_start != Weekday::Mon {
```

Comment:
I am not so sure if we should skip `week_start` and `interval` if they are equal to the default. Yes, they might not have been specified by the user, but they might have been explicitly set as well and expected it to show up in the string. 

It could also be the case that `Weekday` defaults to `Sun` in the future and that could be leading to unexpected behaviour for the user if they have a stringified rrule from a previous version and then deserialize it back to  an rrule with other defaults.

Reply:
I wanted to generate the shortest string.
Based on the RFC, `Mon` and `1` are the default values: https://datatracker.ietf.org/doc/html/rfc5545#section-3.3.10

Reply:
Alright, if they are set as default in the RFC this is a good check to keep the string as short as possible. Thanks for clarifying!